nsv13/code/modules/overmap/fighters/_fighters.dm

```diff
@@ -736,15 +738,21 @@ Been a mess since 2018, we'll fix it someday (probably)
 	relay('nsv13/sound/effects/ship/reactor/gasmask.ogg', "<span class='warning'>The air around you rushes out of the breached canopy!</span>", loop = FALSE, channel = CHANNEL_SHIP_ALERT)
 
 /obj/structure/overmap/small_craft/welder_act(mob/living/user, obj/item/I)
-	. = ..()
+	if(user.a_intent == INTENT_HARM)
+		return FALSE
 	if(obj_integrity >= max_integrity)
 		to_chat(user, "<span class='notice'>[src] isn't in need of repairs.</span>")
-		return FALSE
+		return TRUE
+	if(busy)
+		to_chat(user, "<span class='warning'>Someone's already repairing [src]!</span>")
+		return TRUE
+	busy = TRUE
 	to_chat(user, "<span class='notice'>You start welding some dents out of [src]'s hull...</span>")
-	if(I.use_tool(src, user, 4 SECONDS, volume=100))
+	if(I.use_tool(src, user, ((max_integrity-obj_integrity) / repair_speed) SECONDS, volume=100))
```

Comment:
This seems worse, since you absolutely can't stop until you're done or you lose all your progress.

Reply:
Alright, I'll see if I can figure out how to make it go in increments but automatically, since otherwise it's still just spamclicking or making it repair a large amount at little time cost.

Reply:
Done.